gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -410,6 +436,81 @@ ChannelCredentials createMtlsChannelCredentials() throws IOException, GeneralSec
     return null;
   }
 
+  @VisibleForTesting
+  ChannelCredentials createMtlsToS2AChannelCredentials(
+      InputStream trustBundle, InputStream privateKey, InputStream certChain) throws IOException {
+    if (trustBundle == null || privateKey == null || certChain == null) {
+      return null;
+    }
+    return TlsChannelCredentials.newBuilder()
+        .keyManager(privateKey, certChain)
+        .trustManager(trustBundle)
+        .build();
+  }
+
+  ChannelCredentials createPlaintextToS2AChannelCredentials(String plaintextAddress) {
```

Comment:
Just small, quick comments for these methods too. i.e I am curious why TlsChannelCredentials vs S2AChannelCredentials and why not S2AChannelCredentials for both?

Reply:
The `TlsChannelCredentials` are used to secure channel between S2A when `mtlsAddress` exists, MTLS-MDS creds are there and can be loaded. They are given as a parameter to create `S2AChannelCredentials`:
```
mtlsToS2AChannelCredentials =
          createMtlsToS2AChannelCredentials(trustBundle, privateKey, certChain);
...
return S2AChannelCredentials.newBuilder(mtlsAddress, mtlsToS2AChannelCredentials).build();
```

`S2AChannelCredentials` are used to secure the channel between the client and the service (Google API).

Added some javadocs in https://github.com/googleapis/sdk-platform-java/pull/3326/commits/eb7022524c87eb98b2faf7b2770e080e20bed2c2